packages/shared/src/bounties.ts

```diff
@@ -0,0 +1,292 @@
+import { sendTransaction } from '@acala-network/chopsticks-testing'
+
+import { type Chain, defaultAccountsSr25519 as devAccounts } from '@e2e-test/networks'
+import { setupNetworks } from '@e2e-test/shared'
+
+import { expect } from 'vitest'
+
+import { checkEvents, checkSystemEvents, scheduleInlineCallWithOrigin } from './helpers/index.js'
+import type { RootTestTree } from './types.js'
+
+/// -------
+/// Helpers
+/// -------
+
+/**
+ * Get the current bounty count
+ */
+async function getBountyCount(client: any): Promise<number> {
+  return (await client.api.query.bounties.bountyCount()).toNumber()
+}
+
+/**
+ * Get a bounty by index
+ */
+async function getBounty(client: any, bountyIndex: number): Promise<any | null> {
+  const bounty = await client.api.query.bounties.bounties(bountyIndex)
+  return bounty.isSome ? bounty.unwrap() : null
+}
+
+/**
+ * Get bounty description by index
+ */
+async function getBountyDescription(client: any, bountyIndex: number): Promise<string | null> {
+  const description = await client.api.query.bounties.bountyDescriptions(bountyIndex)
+  return description.isSome ? description.unwrap().toUtf8() : null
+}
+
+/**
+ * Get approved bounties queue
+ */
+async function getBountyApprovals(client: any): Promise<number[]> {
+  const approvals = await client.api.query.bounties.bountyApprovals()
+  return approvals.map((index: any) => index.toNumber())
+}
+
+/**
+ * Setup accounts with funds for testing
+ */
+async function setupTestAccounts(client: any, accounts: string[] = ['alice', 'bob']) {
+  const accountData: any[] = []
+
+  if (accounts.includes('alice')) {
+    accountData.push([[devAccounts.alice.address], { providers: 1, data: { free: 100000000000000n } }])
+  }
+  if (accounts.includes('bob')) {
+    accountData.push([[devAccounts.bob.address], { providers: 1, data: { free: 100000000000000n } }])
+  }
+  if (accounts.includes('charlie')) {
+    accountData.push([[devAccounts.charlie.address], { providers: 1, data: { free: 100000000000000n } }])
+  }
+
+  await client.dev.setStorage({
+    System: {
+      account: accountData,
+    },
+  })
+}
+
+/**
+ * Get bounty index from BountyProposed event
+ */
+async function getBountyIndexFromEvent(client: any): Promise<number> {
+  const [bountyProposedEvent] = (await client.api.query.system.events()).filter(
+    ({ event }: any) => event.section === 'bounties' && event.method === 'BountyProposed',
+  )
+  expect(bountyProposedEvent).toBeDefined()
+  return (bountyProposedEvent.event.data as any).index.toNumber()
```

Comment:
To avoid `any` casting, you can use `assert`.

Example:
https://github.com/open-web3-stack/polkadot-ecosystem-tests/blob/ca38eb5e9b4892054a235305fd00c1a9b733cc8d/packages/shared/src/multisig.ts#L90-L96